### DISCUSSION TITLE
Detecting my robotic lawn mower as a mouse?

### DISCUSSION THREAD

**Discussion by gitlajnk** [blakeblackshear/frigate#7134] [Ask A Question] [answered]
Great job with frigate. Im in the beginning stages but love it already. Anyhow. My 2k camera is detecting my lawnmower (a robotic husqvarna automower) as a mouse. I have detection on person, bird, cat, dog and mouse on this camera. I know you can filter it out with size if it detect like a person (like referenced in this thread https://github.com/blakeblackshear/frigate/discussions/1420) but does the same apply to smaller object than a certain size as well?

The funny thing is that it is somehow not detecting any real mice/rats. I have a rat trap that was activated and also moved so I know this night but there was no detection recorded. Are those two things perhaps correlated? 

Any help appreciated!

Please see my config below:

detectors:
  ov:
    type: openvino
    device: AUTO
    model:
      path: /openvino-model/ssdlite_mobilenet_v2.xml
# 
model:
  width: 300
  height: 300
  input_tensor: nhwc
  input_pixel_format: bgr
  labelmap_path: /openvino-model/coco_91cl_bkgr.txt
#Global Object Settings
objects:
  track:
    - person
    - cat
    - dog
    - bird
    - car
    - mouse
  filters:
    person:
      min_area: 5000
      max_area: 100000
# 
mqtt:
  host: 192.168.1.xx
  user: mqtt-api-user
  password: xxx!
  port: 1883
#
record:
  enabled: True
  retain:
    days: 7
    mode: motion
  events:
    retain:
      default: 14
      mode: active_objects
# 
cameras:
  Carport:
    record:
      enabled: True
      retain:
        days: 5
      events:
        required_zones:
        - Ingång till Uppfart
        - Parkeringsplats
        retain:
          default: 10    
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://xxx3@192.168x:554/onvif1
          input_args: preset-rtsp-udp
          roles:
            - detect
            - record
    zones:
      Parkeringsplats:
        coordinates: 459,424,445,720,581,720,760,720,999,720,1280,720,1280,290,1113,213,878,104,764,104,656,109,554,119
      Ingång till Uppfart:
        coordinates: 407,30,811,21,809,80,698,83,541,95,411,98
    motion:
      mask:
      - 865,0,464,0,404,0,412,94,498,114,548,118,934,103
      - 856,0,1280,0,1280,146,1280,224,1280,326,1051,167,977,122,903,89,856,89,786,85,781,46
    objects:
      track:
      - person
      - cat
      - dog
      - bird
      - mouse
      filters:
        car:
          mask:
            - 524,0,597,0,597,74,528,84

  Uterum:
    detect:
      enabled: true
      width: 1280
      height: 720
      fps: 5
    objects:
      track:
      - person
      - cat
      - dog
      - bird
      - mouse
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://xx@192.x.x54/onvif1
          input_args: preset-rtsp-udp
          roles:
            - detect
            - record
    motion:
      mask:
      - 303,29,746,58,730,206,532,168,310,155                   

  Vardagsrum_kamera_baksida:
    detect:
      enabled: true
      width: 2560
      height: 1440
      fps: 5
    objects:
      track:
      - person
      - cat
      - dog
      - bird
      - mouse
    
    record:
      enabled: True
      retain:
        days: 7
        mode: all
      events:
        retain:
          default: 14
          mode: motion
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://xx:tp123456xxx8.x:554/stream1
          input_args: preset-rtsp-restream
          roles:
            - detect
            - record
    motion:
      mask:
        - 2332,0,0,0,0,621,0,749,119,705,239,915,351,874,370,694,271,495,358,465,513,463,548,233,751,201,890,199,986,185,1004,228,2295,123
        - 1794,1440,2560,1440,2560,869,2414,826,2289,762,2019,764,1965,1070,1876,1212,1830,1317,1759,1319
            
  Vardagsrum_kamera_arne:
    detect:
      enabled: true
      width: 1920
      height: 1080
      fps: 5
    objects:
      track:
        - person
        - cat
        - dog
        - bird
        - mouse
    record:
      enabled: True
      retain:
        days: 7
        mode: all
      events:
        retain:
          default: 14
          mode: motion
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp:/xxxtp123456@xx54/stream2
          input_args: preset-rtsp-restream
          roles:
            - detect
            - record
    motion:
      mask:
      - 0,0,0,630,1058,389,1240,364,1376,327,1443,173,1481,63,1312,27,1174,0,1027,0,845,0
    zones:
       trigger_1_arne:
         coordinates: 945,163,919,536,1006,583,1069,622,1173,685,1329,719,1461,565,1611,498,1679,347,1605,282,1433,236,1290,201
       trigger_2_arne:
         coordinates: 1022,961,849,1051,600,1080,0,1059,0,699,618,350,819,555,1280,744

**Comment by blakeblackshear** [comment, marked as answer]
Mouse in the coco dataset is a computer mouse, not the animal.

**Reply by gitlajnk** [reply]
Thanks so much for your answer. That was a little bit suprising (to me at least). May ask, would it be possible to detect small things like a mouse or rat or the dataset/model is not meant to detect such small objects?

**Reply by blakeblackshear** [reply]
It's possible, but you need a model that was trained to recognize mice/rats. I don't know of any models designed to do it. It may be added to Frigate+ in the future.

**Reply by FrigateEnthusiast** [reply]
Aha! LOL! That explains why when I put a cam in my attic to try to snapshot a rodent heard scampering about every so often, and added "mouse", the only hits I got were a power brick with attached ac cord, laying on the floor of my office.

**Reply by ankycooper** [reply]
It detects my dog as a person!

**Comment by gitlajnk** [comment]
The lawn mower seems difficult for the AI to understand. Now sometimes it is detected as a bird. Some way of mitigating this?
